### ISSUE TITLE
DNS - hosts redirect without root!

### ISSUE THREAD

**Issue by talexop** [LayoutXML/SABS#14]
Hello,
Do you think it would be possible to redirect traffic for specific domain names to a defined IP address?
Like changing the hosts files? This would be great. There is no other app without root that can do that.
Thanks for your effort.

**Comment by LayoutXML** [comment]
Yes, it is possible. Why would you want to redirect traffic elsewhere and not block it completely? I want to understand if the time spent on this issue (a good few hours) is worth it.

**Comment by talexop** [comment]
Generally I agree with you that this is a more advanced setting. For example if I want to enable international  radio station streaming to my Xiaomi aqara gateway I need to change the hosts file (http://ximiraga.ru/#install). From my point of view it would be great to add it. For people that want this function your app will be the only one offering this ;) It is up to you! 
Thank you again for your efford!

**Comment by Mike-BITW** [comment]
On a rooted device it's trivial in /system/etc/hosts to redirect a domain to some other IP besides the traditional 127.0.0.1 and the Knox firewall can do the same thing, but I'm having a difficult time coming up with many real life situations where this would be required ore even simply wise.  I see your example which redirects api.ximalaya.com from 180.153.255.7 (in China?) to 82.146.38.202 (in Russia?) but out of curiosity, why?

**Comment by talexop** [comment]
api.ximalaya.com is used to retrieve the available radio stations. By default the app allows only to choose radio streams from china. Using the IP from Russia give you a lot of international radio stations.
